src/components/fx/hover.js

```diff
@@ -1662,6 +1734,39 @@ function hoverAvoidOverlaps(hoverLabels, axKey, fullLayout) {
     }
 }
 
+function getHoverLabelOffsets(hoverLabel, rotateLabels) {
+    var offsetX = 0;
+    var offsetY = hoverLabel.offset;
+
+    if(rotateLabels) {
+        offsetY *= -YSHIFTY;
+        offsetX = hoverLabel.offset * YSHIFTX;
+    }
+
+    return {
+        x: offsetX,
+        y: offsetY
+    };
+}
+
+function getLabelShiftX(hoverLabel) {
+    var alignShift = {start: 1, end: -1, middle: 0}[hoverLabel.anchor];
+    var txx = alignShift * (HOVERARROWSIZE + HOVERTEXTPAD);
+    var tx2x = txx + alignShift * (hoverLabel.txwidth + HOVERTEXTPAD);
+
+    var isMiddle = hoverLabel.anchor === 'middle';
+    if(isMiddle) {
+        txx -= hoverLabel.tx2width / 2;
+        tx2x += hoverLabel.txwidth / 2 + HOVERTEXTPAD;
+    }
+
+    return {
+        alignShift: alignShift,
+        xx: txx,
+        x2x: tx2x
```

Comment:
Instead of using `xx` and `x2x`, could we possibly improve the naming here?

Reply:
to be honest I am not 100% sure what the original author intended with variable names `txx` and `tx2x`. I guessed `t` might be translation or shift, so I kept the `xx` and `x2x` part with the `shift` part being in the function name. Do you have a suggestion for changing those? :)

Reply:
I had another look at this. Seems `txx` was to mean `x` of text (`tx`) and `tx2x` `x` of text2 (`tx2`). I changed it to more descriptive variable names accordingly.